### PR TITLE
0.9.3 Start sorting merged instanced meshes by size

### PR DESCRIPTION
**Pull request by Verkhovskaya**

### CODE REVIEW

**Comment by anderscognite** [comment]
I think we should have an easier to read sorting, and never return of type `any`. Can you prove that this way of sorting is significantly faster than creating objects with the information we need, and sort them?

**Comment by Verkhovskaya** [comment]
It's about ~0.2 seconds slower if we use .sort. Switched the code to .sort().

File: src/optimizations/mergeInstancedMeshes.ts [hunk 1]
```diff
@@ -1,61 +1,60 @@
...
-  let numMappings = 0;
-  collections.forEach(collection => {
-    const triangleCount = collection.mappings.count * collection.triangleCount;
-    if (triangleCount < TRIANGLE_COUNT_LIMIT) {
-      numMappings += collection.mappings.count;
-    }
-  });
-  return numMappings;
-}
+type MappingInfo = {collectionIndex: number, mappingIndex: number, size: number};
```
Comment (anderscognite) [inline src/optimizations/mergeInstancedMeshes.ts:11]: MappingInfo or MappingData? You typically don't sort info, but data? Or maybe I'm being silly?

File: src/optimizations/mergeInstancedMeshes.ts [hunk 2]
```diff
@@ -1,61 +1,60 @@
...
 
   for (const sector of rootSector.traverseSectorsBreadthFirst()) {
+    // for each instanced mesh object with a given fileId
     sector.instancedMeshGroup.meshes.forEach(instancedMesh => {
+      // Get all small instanced mesh collections, sorted by size
+      const smallCollections = instancedMesh.collections.filter(
+        collection => (collection.mappings.count * collection.triangleCount <= TRIANGLE_COUNT_LIMIT));
+      const mappingsSortedBySize: MappingInfo[] = [];
+      smallCollections.forEach((collection, collectionIndex) => {
+        for (let i = 0; i < collection.mappings.capacity; i++) {
```
Comment (anderscognite) [inline src/optimizations/mergeInstancedMeshes.ts]:
```suggestion
        for (let mappingIndex = 0; mappingIndex < collection.mappings.capacity; mappingIndex++) {
```

File: src/optimizations/mergeInstancedMeshes.ts [hunk 3]
```diff
@@ -1,61 +1,60 @@
...
+          mappingsSortedBySize.push({
+            collectionIndex: collectionIndex, mappingIndex: i, size: collection.mappings.getSize(i) });
```
Comment (anderscognite) [inline src/optimizations/mergeInstancedMeshes.ts]:
```suggestion
            collectionIndex, mappingIndex, size: collection.mappings.getSize(i) });
```

**Review by anderscognite — CHANGES_REQUESTED** [review]
Looks good! I have a couple of suggestions, what do you think?

File: src/optimizations/mergeInstancedMeshes.ts (hunk 1 above)
Comment (Verkhovskaya) [reply src/optimizations/mergeInstancedMeshes.ts:11]: Personally, I only call things 'Data' if I'm going to be keeping it around for a while.

Comment (Verkhovskaya) [reply src/optimizations/mergeInstancedMeshes.ts:11]: Google to the rescue 😂. "Data are simply facts or figures — bits of information, but not information itself. When data are processed, interpreted, organized, structured or presented so as to make them meaningful or useful, they are called information. Information provides context for data.". ... I'd say it's info?

Comment (Verkhovskaya) [reply src/optimizations/mergeInstancedMeshes.ts:11]: Since I'm technically processing the data to make it useful

Comment (Verkhovskaya) [reply src/optimizations/mergeInstancedMeshes.ts:11]: Like, I'd never stick this in a database, so it's not data

Comment (anderscognite) [reply src/optimizations/mergeInstancedMeshes.ts:11]: Sure!

**Comment by anderscognite** [comment]
Travis fails :P

**Comment by Verkhovskaya** [comment]
Oops, fixed. One of your suggested edits had a typo, I should have caught that

**Review by anderscognite — APPROVED** [review]
I was going to ask about the order :D

**Pull request merged**